src/connectivity.jl

```diff
@@ -536,17 +536,33 @@ end
 """
     isgraphical(degs)
 
-Return true if the degree sequence `degs` is graphical, according to
-[Erdös-Gallai condition](http://mathworld.wolfram.com/GraphicSequence.html).
+Return true if the degree sequence `degs` is graphical.
+A sequence of integers is called graphical, if there exists a graph where the degrees of its vertices form that same sequence.
 
 ### Performance
-    Time complexity: ``\\mathcal{O}(|degs|^2)``
+Time complexity: ``\\mathcal{O}(|degs|*\\log(|degs|))``.
+
+### Implementation Notes
+According to Erdös-Gallai theorem, a degree sequence ``\\{d_1, ...,d_n\\}`` (sorted in descending order) is graphic iff the sum of vertex degrees is even and the sequence obeys the property -
+```math
+\\sum_{i=1}^{r} d_i \\leq r(r-1) + \\sum_{i=r+1}^n min(r,d_i)
+```
+for each integer r <= n-1
 """
-function isgraphical(degs::Vector{Int})
+function isgraphical(degs::Vector{<:Integer})
     iseven(sum(degs)) || return false
-    n = length(degs)
-    for r = 1:(n - 1)
-        cond = sum(i -> degs[i], 1:r) <= r * (r - 1) + sum(i -> min(r, degs[i]), (r + 1):n)
+    sorted_degs = sort(degs, rev = true)
+    n = length(sorted_degs)
+    cur_sum = 0
```

Comment:
`cur_sum` = zero(UInt64)`, otherwise we will have some problems with type instability later, as the type of `cur_sum` changes from `Int` to `UInt64` in the loop.